src/components/features/Home/GoodsRankingSection/index.tsx

```diff
@@ -14,15 +16,32 @@ export const GoodsRankingSection = () => {
     targetType: 'ALL',
     rankType: 'MANY_WISH',
   });
+  const {
+    data = { products: [] },
+    isLoading,
+    isError,
+  } = useQuery({
+    queryKey: ['rankingSection', filterOption.targetType, filterOption.rankType],
+    queryFn: () => fetchRankingSection(filterOption.targetType, filterOption.rankType),
+  });
+
+  const goodsList = data?.products ?? [];
 
-  // GoodsMockData를 21번 반복 생성
+  let currentStatus;
+  if (isLoading) currentStatus = <StatusDiv>로딩중...</StatusDiv>;
+  if (isError) currentStatus = <StatusDiv>데이터를 불러오는 중에 문제가 발생했습니다.</StatusDiv>;
+  if (goodsList.length === 0) currentStatus = <StatusDiv>보여줄 상품이 없어요!</StatusDiv>;
```

Comment:
```suggestion
  const statusElem = (() => {
      if (isLoading) currentStatus = <StatusDiv>로딩중...</StatusDiv>;
      if (isError) currentStatus = <StatusDiv>데이터를 불러오는 중에 문제가 발생했습니다.</StatusDiv>;
      if (goodsList.length === 0) currentStatus = <StatusDiv>보여줄 상품이 없어요!</StatusDiv>;
  })()
```

보통 불변성을 최대한 지켜서 코드를 작성하는게 더 안전해서 저는 개인적으론 let은 되도록이면 안쓰려고 하긴 합니다 ㅎㅎ